packages/matchbox/src/components/TextField/TextField.js

```diff
@@ -1,179 +1,207 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import { Label } from '../Label';
 import { Error } from '../Error';
 import { Connect } from '../Connect';
-import classnames from 'classnames';
-import styles from './TextField.module.scss';
-
-class TextField extends Component {
-  static displayName = 'TextField';
-
-  static propTypes = {
-    /**
-     * 'left' | 'center' | 'right'
-     */
-    align: PropTypes.oneOf(['left', 'center', 'right']),
-    id: PropTypes.string,
-    multiline: PropTypes.bool,
-    /**
-     * 'text' | 'email' | 'number' | 'password' | 'search' | 'tel' | 'url' | 'date'
-     */
-    type: PropTypes.oneOf(['text', 'email', 'number', 'password', 'search', 'tel', 'url', 'date']),
-    placeholder: PropTypes.string,
-    autoFocus: PropTypes.bool,
-    disabled: PropTypes.bool,
-    readOnly: PropTypes.bool,
-    required: PropTypes.bool,
-    /**
-     * 'none' | 'both' | 'horizontal' | 'vertical' | 'block' | 'inline'
-     */
-    resize: PropTypes.oneOf(['none', 'both', 'horizontal', 'vertical', 'block', 'inline']),
-    label: PropTypes.string,
-    labelHidden: PropTypes.bool,
-    helpText: PropTypes.node,
-    connectLeft: PropTypes.node,
-    connectRight: PropTypes.node,
-    prefix: PropTypes.node,
-    suffix: PropTypes.node,
-    error: PropTypes.string,
-    errorInLabel: PropTypes.bool,
-    onChange: PropTypes.func,
-    onFocus: PropTypes.func,
-    onBlur: PropTypes.func
-  };
-
-  static defaultProps = {
-    resize: 'both',
-    type: 'text'
-  };
-
-  state = {
-    paddingLeft: null,
-    paddingRight: null
-  };
-
-  componentDidMount() {
-    this.calculatePadding();
-  }
-
-  componentDidUpdate({ prefix, suffix }) {
-    if (prefix !== this.props.prefix || suffix !== this.props.suffix) {
-      this.calculatePadding();
-    }
+import { Box } from '../Box';
+import { HelpText } from '../HelpText';
+import { getRectFor, roundToBaseline } from '../../helpers/geometry';
+import useInputDescribedBy from '../../hooks/useInputDescribedBy';
+import { tokens } from '@sparkpost/design-tokens';
+import { margin } from 'styled-system';
+import styled from 'styled-components';
+import { pick, omit } from '@styled-system/props';
+
+const StyledWrapper = styled('div')`
+  ${margin}
+`;
+
+// TODO Make this reusable and shared with Select
+const FieldBox = props => {
+  return (
+    <Box
+      as="input"
+      disabled={props.disabled}
```

Comment:
Just like the `<Select/>`, it would be good to use the `required` attribute when the `required` prop is true. Appears to be missing at the moment:

<img width="1389" alt="Screen Shot 2020-03-11 at 12 32 53 PM" src="https://user-images.githubusercontent.com/3613392/76440648-896aaa00-6394-11ea-86c7-974e7c0e89c2.png">

Reply:
Looks like it's just a missing `id` prop in the Storybook story.

Technically the `id` prop should be required - what would you think about making it required? Could force devs to pass in one for each component instance, improving a11y on the app pretty tremendously fairly easily to boot.

Reply:
Sure, if we think it wont be too much of a hassle at this point. I think most TextFields use the redux-form wrapper which handles id nicely.

Reply:
👍 I added it to the story. It shouldn't be a big hassle.